guides/assets/javascripts/guides.js

```diff
@@ -102,5 +103,83 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var anchorLinks = Array.from(document.querySelectorAll(".anchorlink"));
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
+      removeHighlight();
+      if (!elem) return;
+      elem.classList.add("active");
+      if (window.innerWidth > DESKTOP_THRESHOLD) {
+        elem.scrollIntoView({ block: 'center', inline: 'end' });
+      }
+    }
+
+    var resetHighlight = function () {
+      var chapters = document.querySelector(".chapters");
+      chapters?.scroll({ top: 0 });
+      removeHighlight();
+    }
+
+    var matchingNavLink = function (link) {
+      if (link) {
+        return document.querySelector('#subCol [href="' + link.getAttribute("href") + '"]');
+      }
+      return null;
+    }
+
+    var scrollingUp = function (i) {
+      return i.boundingClientRect.bottom < (i.rootBounds.bottom + i.rootBounds.top) / 2;
+    }
+
+    var prevLink = function (link) {
+      var index = anchorLinks.indexOf(link);
+      if (index === 0) {
+        return null;
+      }
+      return matchingNavLink(anchorLinks[index - 1]);
+    }
+
+    var linkToHighlight = function (i) {
+      if (scrollingUp(i)) {
+        return prevLink(i.target);
+      } 
+      return matchingNavLink(i.target);
+    }
+
+    var navHighlight = function (entries) {
+      entries.forEach(function (entry) {
+        if (entry.isIntersecting) {
+          updateHighlight(linkToHighlight(entry));
+        }
+      });
+    }
+
+    var threshold = parseInt(window.innerHeight - 50);
```

Comment:
Doesn't `window.innerHeight - 50` always return an integer? Do we need `parseInt`?